include/ensmallen_bits/cmaes/cmaparameters.hpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file cmaesparametes.hpp
+ * @author John Hoang
+ * @author Marcus Edel 
+ * 
+ * Definition of the Covariance Matrix Adaptation Evolution Strategy as proposed
+ * by N. Hansen et al. in "Completely Derandomized Self-Adaptation in Evolution
+ * Strategies".
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_CMAES_CMA_PARAMETERS_HPP
+#define ENSMALLEN_CMAES_CMA_PARAMETERS_HPP
+
+namespace ens{
```

Comment:
I was thinking instead of a parameters class, we could do something similar as we did for the SGD optimizer class, which acts as a baseline for other optimizers like Adam.

https://github.com/mlpack/ensmallen/blob/master/include/ensmallen_bits/adam/adam.hpp#L201

instantiates the SGD base optimizer which we use to call the Optimize function. We could call any function, e.g. let's say the base class implements a function to mutate the population, we could call that function in the extended optimizer as well, as long as the method is public.